File: backend/events/deleteEvent.mjs
import { GetCommand, DeleteCommand } from "@aws-sdk/lib-dynamodb";
import { db, EVENTS_TABLE } from "./shared/db.mjs";
import { ok, badRequest, notFound, forbidden, serverError, cors } from "./shared/response.mjs";
import { audit, callerFromEvent, isAdmin } from "./shared/auth.mjs";

export async function handler(event) {
  if (event.httpMethod === "OPTIONS") return cors();

  const caller = callerFromEvent(event);
  if (!isAdmin(caller.groups)) return forbidden();

  const { eventId } = event.pathParameters ?? {};
  if (!eventId) return badRequest("eventId is required");

  try {
    const existing = await db.send(new GetCommand({ TableName: EVENTS_TABLE, Key: { eventId } }));
    if (!existing.Item) return notFound("Event not found");

    await db.send(new DeleteCommand({ TableName: EVENTS_TABLE, Key: { eventId } }));
    await audit("event.deleted", {
      entity: "event",
      entityId: eventId,
      actorId: caller.actorId,
      actorLabel: caller.actorLabel,
      meta: { name: existing.Item.name },
    });
    return ok({ deleted: true, eventId });
  } catch (e) {
    console.error(e);
    return serverError();
  }
}

File: backend/events/deleteEvent.py
import logging

from shared.auth import audit, caller_from_event, is_admin
from shared.db import events_table
from shared.response import bad_request, cors, forbidden, not_found, ok, server_error, with_cors

log = logging.getLogger()


@with_cors
def handler(event, context):
    if event.get("httpMethod") == "OPTIONS":
        return cors()

    caller = caller_from_event(event)
    if not is_admin(caller["groups"]):
        return forbidden()

    event_id = (event.get("pathParameters") or {}).get("eventId")
    if not event_id:
        return bad_request("eventId is required")

    try:
        existing = events_table().get_item(Key={"eventId": event_id}).get("Item")
        if not existing:
            return not_found("Event not found")

        events_table().delete_item(Key={"eventId": event_id})
        audit("event.deleted", entity="event", entity_id=event_id,
              actor_id=caller["actorId"], actor_label=caller["actorLabel"],
              meta={"name": existing.get("name")})
        return ok({"deleted": True, "eventId": event_id})
    except Exception:
        log.exception("deleteEvent failed")
        return server_error()
